base/collection_n1ql_common.go

```diff
@@ -44,7 +44,7 @@ type N1QLStore interface {
 	ExplainQuery(ctx context.Context, statement string, params map[string]interface{}) (plan map[string]interface{}, err error)
 	GetIndexMeta(ctx context.Context, indexName string) (exists bool, meta *IndexMeta, err error)
 	Query(ctx context.Context, statement string, params map[string]interface{}, consistency ConsistencyMode, adhoc bool) (results sgbucket.QueryResultIterator, err error)
-	IsErrNoResults(error) bool
+	IsErrNoResults(context.Context, error) bool
```

Comment:
I don't know if it makes sense to add context to this API, just for warn logging in the leaky bucket conversion step.  Given that we wouldn't have been able to get to the point where we would need to check this (i.e. have results from executeQuery) for a non-N1QL data store, it doesn't seem needed?